plugins/kinetic-dp/fu-kinetic-dp-puma-aux-isp.c

```diff
@@ -0,0 +1,847 @@
+						   1,
+						   error)) {
+			g_prefix_error(error,
+				       "failed to read PUMA_DPCD_SINK_MODE_REG for status: ");
+			g_set_error_literal(error,
+					    FWUPD_ERROR,
+					    FWUPD_ERROR_INTERNAL,
+					    "waiting for PUMA_FW_UPDATE_DONE failed.");
+			return FALSE;
+		}
+		if (status == PUMA_FW_UPDATE_DONE) {
+			g_debug("Firmware Update Done");
+			break;
+		} else {
+			/* wait interval before check the status again */
+			g_usleep(interval_ms * 1000);
+			if (max_time_ms >= interval_ms)
+				max_time_ms -= interval_ms;
+			else
+				max_time_ms = 0;
+		}
+	}
+	/* time out */
+	if (max_time_ms == 0) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "waiting for PUMA_FW_UPDATE_READY failed.");
+		return FALSE;
+	} else {
+		return TRUE;
+	}
+}
+
+static gboolean
+fu_kinetic_dp_puma_aux_isp_get_device_info(FuKineticDpAuxIsp *self,
+					   FuKineticDpDevice *device,
+					   KtDpDevInfo *dev_info,
+					   GError **error)
+{
+	g_autoptr(FuKineticDpConnection) connection = NULL;
+	guint8 dpcd_buf[16] = {0};
+
+	connection = fu_kinetic_dp_connection_new(fu_udev_device_get_fd(FU_UDEV_DEVICE(device)));
+
+	/* chip ID, FW work state, and branch ID string are known */
+	if (!fu_kinetic_dp_connection_read(connection,
+					   DPCD_ADDR_BRANCH_FW_SUB,
+					   dpcd_buf,
+					   sizeof(dpcd_buf),
+					   error)) {
+		g_prefix_error(error, "reading branch id failed: ");
+		return FALSE;
+	}
+	dev_info->chip_rev = dpcd_buf[1]; /* DPCD 0x509 HW_Ver */
+	/* DPCD 0x50A,0x50B,0x508 */
+	dev_info->fw_info.std_fw_ver =
+	    (guint32)dpcd_buf[2] << 16 | (guint32)dpcd_buf[3] << 8 | dpcd_buf[0];
+
+	/* TODO: implement Gprobe over Aux to read active flash bank */
+
+	dev_info->fw_info.boot_code_ver = 0;
+	/* TODO: Add function to read CMDB information */
+	dev_info->fw_info.std_cmdb_ver = 0;
+	dev_info->fw_info.cmdb_rev = 0;
+
+	return TRUE;
+}
+
+/* starting puma isp process */
+static gboolean
+fu_kinetic_dp_puma_aux_isp_start(FuKineticDpAuxIsp *self,
+				 FuKineticDpDevice *device,
+				 FuFirmware *firmware,
+				 FuProgress *progress,
+				 const KtDpDevInfo *dev_info,
+				 GError **error)
+{
+	FuKineticDpPumaAuxIsp *puma_aux_isp = FU_KINETIC_DP_PUMA_AUX_ISP(self);
+	FuKineticDpPumaAuxIspPrivate *priv = GET_PRIVATE(puma_aux_isp);
+	FuKineticDpFirmware *firmware_self = FU_KINETIC_DP_FIRMWARE(firmware);
+	gboolean is_app_mode = (KT_FW_STATE_RUN_APP == dev_info->fw_run_state) ? TRUE : FALSE;
+	const guint8 *payload_data;
+	gsize payload_len;
+	gboolean retcode = FALSE;
+	guint8 cmd;
+	g_autoptr(FuKineticDpConnection) connection = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(FuFirmware) img = NULL;
+	g_autoptr(GBytes) app = NULL;
+	g_autoptr(GBytes) isp_drv = NULL;
+	guint8 mca_oui[DPCD_SIZE_IEEE_OUI] = {MCA_OUI_BYTE_0, MCA_OUI_BYTE_1, MCA_OUI_BYTE_2};
+
+	/* progress */
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_flag(progress, FU_PROGRESS_FLAG_GUESSED);
```

Comment:
could you get the actual values? the daemon will print them out on the CLI with --verbose